### PR TITLE
Remove NaNs from loss in LRFinder

### LINKED ISSUE #1850: lr_find doesn't return the correct suggestion if some losses are nan
## 🐛 Bug

lr_finder doesn't return the correct suggestion if some losses are `nan`. The returned loss is the one corresponding to the `nan` value, which is very big in my case.

image

### To Reproduce

This depends on the dataset. Please see the code sample.

#### Code sample

I believe this is caused by numpy. The related code should be https://github.com/PyTorchLightning/pytorch-lightning/blob/b84b02400a312240a6429c186cc63514eeb45a82/pytorch_lightning/trainer/lr_finder.py#L325

```python
example_losses = [0.90, 0.89, 0.87, 0.86, 0.85, 0.84]
print(np.gradient(example_losses).argmin())
example_losses = [0.90, 0.89, 0.87, 0.86, 0.85, 0.84, float('nan')]
print(np.gradient(example_losses).argmin())
```
Output:
```
1
5
```
### Expected behavior

Return the correct suggested loss.

### Environment

* CUDA:
	- GPU:
	- available:         False
	- version:           10.2
* Packages:
	- numpy:             1.18.4
	- pyTorch_debug:     False
	- pyTorch_version:   1.5.0
	- pytorch-lightning: 0.7.6
	- tensorboard:       2.2.0
	- tqdm:              4.45.0
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         
	- python:            3.7.6
	- version:           #1 SMP Debian 4.19.118-2 (2020-04-29)

### Additional context

NA

### PR DESCRIPTION
**Pull request by rohitgr7**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?   
- [x] Did you write any new necessary tests?  
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## What does this PR do?
Fixes #1850.

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/trainer/lr_finder.py [hunk 1]
```diff
@@ -321,8 +321,9 @@ def suggestion(self, skip_begin: int = 10, skip_end: int = 1):
 
         """
         try:
-            loss = self.results["loss"][skip_begin:-skip_end]
-            min_grad = (np.gradient(np.array(loss))).argmin()
+            loss = np.array(self.results["loss"][skip_begin:-skip_end])
+            loss = loss[~np.isnan(loss)]
```
Comment (awaelchli) [inline pytorch_lightning/trainer/lr_finder.py:325]: is perhaps [torch.isfinite]( https://pytorch.org/docs/stable/torch.html#torch.isfinite) a better option here? it would also include +/- inf values. What do you think?

Comment (rohitgr7) [reply pytorch_lightning/trainer/lr_finder.py:325]: Yep. torch.isfinite is a better option here. Let me change it.

Comment (awaelchli) [reply pytorch_lightning/trainer/lr_finder.py]: nice. numpy works too :)

**Comment by awaelchli** [comment]
notifiying also @SkafteNicki

**Comment by SkafteNicki** [comment]
`Nan` can happens at the very end of the search if the loss/gradient explodes too much before we stop the search. I think we could add a simple test, something like
```
lr_finder = trainer.lr_find(...)
lr_before_nan = lr_finder.suggestion()
lr_finder.results['loss'][-1] = np.nan
lr_after_nan = lr_finder.suggestion()
assert lr_before_nan == lr_after_nan
```
Do simple run, run the suggestion algorithm, introduce a nan artificially, recompute suggestion. If this works the two suggestions should be the same.

**Comment by Borda** [comment]
just a note for speed, could we do the filtering in PyTorch not converting to NumPy?

**Comment by SkafteNicki** [comment]
@borda the values are stored as a list (of floats), so i guess that the penalty of converting to numpy vs pytorch is small

**Comment by williamFalcon** [comment]
the conversion is at the full end of the lr finder and before training begins no?
this means the cost is negligible unless i'm missing something

**Comment by williamFalcon** [comment]
Is this ready?

**Comment by rohitgr7** [comment]
> Is this ready?

I will add the test case now.
